File: 프로그래머스/2/140107. 점 찍기/점 찍기.java
class Solution {
    public long solution(int k, int d) {
        long answer = 0;

        for (long x = 0; x <= d; x += k) {
            // 가능한 y의 최대 거리 계산 (원 내부 조건)
            long maxY = (long) Math.sqrt((long) d * d - x * x);
            // y는 0, k, 2k, ... maxY 이하까지 가능하므로 개수는 (maxY / k) + 1
            answer += (maxY / k) + 1;
        }

        return answer;
    }
}

File: 프로그래머스/2/140107. 점 찍기/점 찍기.kt
import kotlin.math.sqrt
import kotlin.math.floor

class Solution {
    fun solution(k: Int, d: Int): Long {
        var answer = 0L

        var x = 0L
        while (x <= d) {
            // 가능한 y의 최대 거리 계산
            val maxY = floor(sqrt((d.toLong() * d - x * x).toDouble())).toLong()
            // y는 0, k, 2k, ... maxY 이하이므로 개수는 (maxY / k) + 1
            answer += (maxY / k) + 1
            x += k
        }

        return answer
    }
}
